CRM/UF/Page/Field.php

```diff
@@ -126,7 +126,9 @@ public function browse() {
     $this->assign('isGroupReserved', $isGroupReserved);
 
     $profileType = CRM_Core_BAO_UFField::getProfileType($this->_gid);
-    if ($profileType == 'Contribution' || $profileType == 'Membership' || $profileType == 'Activity' || $profileType == 'Participant') {
```

Comment:
You might wanna use _checkProfileType_ method for checking whether the profile is mixed or not.